DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/matmul/src/matmul.hpp

```diff
@@ -0,0 +1,143 @@
+#ifndef __MATMUL_HPP__
+#define __MATMUL_HPP__
+
+#include <iostream>
+
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/interfaces.hpp>
+#include <sycl/sycl.hpp>
+
+#include "memory_transfers.hpp"
+// Included from DirectProgramming/C++SYCL_FPGA/include/
+#include "streaming_matmul.hpp"
+
+#if not defined(IS_BSP)
+using sycl::ext::intel::experimental::property::usm::buffer_location;
+#endif
+
+// Forward declare the kernel and pipe names
+// (This prevents unwanted name mangling in the optimization report.)
+class FeederA;
+class FeederB;
+class Matmul;
+class Drain;
+class APipe;
+class BPipe;
+class CPipe;
+class DonePipe;
+
+/**
+ * Implementation of the matrix multiplication using multiple streaming kernels.
+ * Parameterized by datatype, matrix size, and tile size. Exercises the kernels
+ * by running multiple repetitions for a set of matrices.
+ *
+ * Function arguments:
+ *  q: device queue
+ *  a_matrix: input matrix pointer (given in column-major)
+ *  b_matrix: input matrix pointer (given in row-major, i.e., transposed)
+ *  c_matrix: output matrix pointer (will be stored in column-major)
+ *  repetitions: number of repetitions of the computation to execute
+ *
+ */
+template <typename TT,          // Datatype of the elements of the matrix
+          int k_rows_a,         // Rows of matrix A
```

Comment:
I don't think there is a "k_" convention that we have used before.   In this case it might be better to just use "rows_a" instead of "k_rows_a".   You are already using the kMatsizeA convention correctly later in the example.

Reply:
That is true. Some of the new FPGA samples started using k_ for template parameters in order to improve readability: when in the middle of the function it's hard to know what is a template parameter, and therefore there is no indication that the variable is constant or not. Having this k_ helps.
But if you think that we should not use that, we can come back to the usual naming convention

Reply:
Updated the sample to use the usual naming convention for template parameters.